src/value/value.rs

```diff
@@ -374,7 +374,53 @@ pub enum ObjectKey {
     Null,
 }
 
+impl std::cmp::Ord for ObjectKey {
+    fn cmp(&self, other: &ObjectKey) -> std::cmp::Ordering {
+        self.canonical_sort_key().cmp(&other.canonical_sort_key())
+    }
+}
+
+impl std::cmp::PartialOrd for ObjectKey {
+    fn partial_cmp(&self, other: &ObjectKey) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
 impl ObjectKey {
+    // Returns a key to sort `ObjectKey`s by major type, then byte-length, then lexicographically
+    // by content. Equivalent to sorting `ObjectKey`s __lexicographically__ by their CBOR serialization.
+    //
+    // Note: the first guidelines regarding canonicalized CBOR
```

Comment:
Why was the canonical order changed?

Reply:
Some of the discussion regarding the change is located [here](https://www.ietf.org/mail-archive/web/cbor/current/msg00264.html), but mainly for implementation simplicity in that values of different types do not need to be compared.